src/_types.ts

```diff
@@ -21,3 +21,10 @@ export type NonEmptyArray<T> = [T, ...Array<T>];
  * @see This was inspired by the type-definition of Promise.all (https://github.com/microsoft/TypeScript/blob/1df5717b120cddd325deab8b0f2b2c3eecaf2b01/src/lib/es2015.promise.d.ts#L21)
  */
 export type IterableContainer<T = unknown> = ReadonlyArray<T> | [];
+
+// Inspired and largely copied from `sindresorhus/ts-extras`:
+// @see https://github.com/sindresorhus/ts-extras/blob/44f57392c5f027268330771996c4fdf9260b22d6/source/object-keys.ts
+export type ObjectKeys<T extends Record<PropertyKey, unknown>> = `${Exclude<
```

Comment:
Any reason to switch from `object` to `Record<PropertyKey, unknown>`. If the typing works for the objects we want, I don't mind us being broader and allowing unusual usages to be applicable too (and if they are buggy, to get an issue report).

https://www.typescriptlang.org/play?#code/C4TwDgpgBA8gRgKwgY2AaQiAzjANgEwB4AVKCAD2AgDt8soB7RFYAPigF4oADAEgG8AouWS4ArvgiEA1pgYAzKMQA0ULCAC2cBrlYBfbgG4AUKEixmqDNgByEAO4kylGnSgAlFAwBORAAreDJDeoNaqYtTS1Az21KzsXHxCIuKSMnKKKmqa2roGJsam4NAAwgCGuLhlcLjQXPzGUFAAFACUAFxQAEQAFhCVDF0mTWWdXfY+BEPGeoXIDNRYwFDkneWV1bWcFkioAHRlWFgAlgDm1M1tnOy9-biDqvxQo90T3lPP9POLwHqtBd8llBvBAsGJcMsuOQ2iYAPSwppNAB6AH5jICdBA9vdTs1yAd-sZ4YiSaTUYV0QsgSBtvBdsA9rJsHj-lBiRx2ABtADkZW5AF1CmZoNYcPJ1lUahA8PhaZZ0JgcARCBLNhBWHCEci0ULilBRTBxRVJbU7PY5fTRWaVca1RrCkA

```suggestion
export type ObjectKeys<T extends object> = `${Exclude<
```

Reply:
It doesn't matter at the moment, since the T sent always matches `extends Record<PropertyKey, unknown>`. But it might be useful in the future.